pkg/registry/ecr/registry.go

```diff
@@ -0,0 +1,136 @@
+
+	if err := r.Enrich(); err != nil {
+		return errors.Wrap(err, "Failed to enrich ECR Registry")
+	}
+
+	if err := r.Validate(); err != nil {
+		return errors.Wrap(err, "Failed to validate ECR Registry")
+	}
+
+	return nil
+}
+
+func (r *Registry) Enrich() error {
+
+	// parse aws credentials
+	var awsCreds registry.AWSCreds
+	if err := json.Unmarshal([]byte(r.Creds), &awsCreds); err != nil {
+		r.Logger.WarnWith("Failed to parse json AWS credentials, checking env", "err", err.Error())
+	}
+
+	awsCreds.Region = util.GetFirstNonEmptyString([]string{awsCreds.Region, strings.TrimSpace(os.Getenv("AWS_DEFAULT_REGION"))})
+	awsCreds.AccessKeyID = util.GetFirstNonEmptyString([]string{awsCreds.AccessKeyID, strings.TrimSpace(os.Getenv("AWS_ACCESS_KEY_ID"))})
+	awsCreds.SecretAccessKey = util.GetFirstNonEmptyString([]string{awsCreds.SecretAccessKey, strings.TrimSpace(os.Getenv("AWS_SECRET_ACCESS_KEY"))})
+	awsCreds.AssumeRole = util.GetFirstNonEmptyString([]string{awsCreds.AssumeRole, strings.TrimSpace(os.Getenv("AWS_ROLE_ARN"))})
+	r.awsCreds = awsCreds
+
+	return nil
+}
+
+func (r *Registry) Validate() error {
+	if err := r.Registry.ValidateParameters(); err != nil {
+		return errors.Wrap(err, "Failed to validate base parameters")
+	}
+
+	if r.awsCreds.Region == "" {
+		return errors.New("AWS Region is required")
+	}
+
+	if r.awsCreds.AccessKeyID == "" {
+		return errors.New("AWS Access Key ID is required")
+	}
+
+	if r.awsCreds.SecretAccessKey == "" {
+		return errors.New("AWS Secret Access Key is required")
+	}
+
+	return nil
+}
+
+func (r *Registry) GetAuthToken(ctx context.Context) (*registry.Token, error) {
+	ecrClient := r.createECRClient(ctx)
+
+	r.Logger.DebugWithCtx(ctx, "Getting authorization token", "SecretName", r.SecretName, "Namespace", r.Namespace)
+	resp, err := ecrClient.GetAuthorizationToken(&ecr.GetAuthorizationTokenInput{})
+
+	if err != nil {
+		r.Logger.WarnWithCtx(ctx, "Failed to get ECR authorization token", "error", err.Error())
+		return nil, errors.Wrap(err, "Failed to get authorization token from ecr client")
+	}
+	r.Logger.DebugWithCtx(ctx, "Got GetAuthorizationToken response from ECR")
+
+	// AuthorizationData is a list as it used to return a token per registry, that is now deprecated.
+	// The returned token can be used to access any Amazon ECR registry that the IAM principal has access to.
+	for _, auth := range resp.AuthorizationData {
+		token := &registry.Token{
+			SecretName:  r.SecretName,
+			Namespace:   r.Namespace,
+			AccessToken: *auth.AuthorizationToken,
+			RegistryUri: r.RegistryUri,
+		}
+		r.Logger.InfoWithCtx(ctx, "Got authorization token", "ExpiresAt", auth.ExpiresAt)
+		return token, nil
+	}
+
+	return nil, errors.New("Failed to retrieve ECR access token")
+}
+
+func (r *Registry) createECRClient(ctx context.Context) *ecr.ECR {
+	r.Logger.DebugWithCtx(ctx, "Creating ECR Client", "region", r.awsCreds.Region, "assumeRole", r.awsCreds.AssumeRole)
+	sess := session.Must(session.NewSession(&aws.Config{
```

Comment:
```suggestion
	sessionInstance := session.Must(session.NewSession(&aws.Config{
```

if you have a collision between the instance and the pkg, add `Instance` as a convention